server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WhiteListManager.java

```diff
@@ -752,9 +754,16 @@ private void initSqlQueries(Connection conn, org.apache.mailet.MailetContext mai
             }
 
             /*
-      Holds value of property sqlFile.
-     */
-            File sqlFile = new File((String) mailetContext.getAttribute("confDir"), "sqlResources.xml").getCanonicalFile();
+                Holds value of property sqlFile.
+            */
+            String confDir = mailetContext
+                    .getAttribute(AttributeName.of("confDir"))
+                    .orElseThrow(() -> new MailetException("Mailet has no 'confDir' attribute"))
+                    .getValue()
+                    .value()
+                    .toString();
```

Comment:
Are you sure you want to trust _toString_?

Reply:
We were trusting `(String)`, it is not equivalent?

Reply:
toString is supposed to be for debug purpose. If you need API stability for a string representation of your object, you should define a `asString` method.